### PR TITLE
[GAP_pkg_curlinterface] new package

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Comment by fingolfin** [comment]
And another failure on x86-linux-musl... Looking at config.log:
```
configure:3541: checking for curl-config
configure:3564: found /workspace/destdir/bin/curl-config
configure:3577: result: /workspace/destdir/bin/curl-config
configure:3635: checking for the version of libcurl
configure:3644: result: 8.11.1
configure:3707: checking whether libcurl is usable
configure:3742: cc -o conftest -g -O2 -I/workspace/destdir/include -I/workspace/destdir/incl
ude  conftest.c -L/workspace/destdir/lib -lcurl  >&5
/workspace/destdir/lib/libcurl.so: undefined reference to `nghttp2_option_set_no_rfc9113_leading_and_trailing_ws_validation'
collect2: error: ld returned 1 exit status
configure:3742: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "curlInterface"
| #define PACKAGE_TARNAME "curlinterface"
| #define PACKAGE_VERSION "GAP package"
| #define PACKAGE_STRING "curlInterface GAP package"
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| /* end confdefs.h.  */
| #include <curl/curl.h>
| int
| main (void)
| {
|
| /* Try and use a few common options to force a failure if we are
|    missing symbols or can't link. */
| int x;
| curl_easy_setopt(NULL,CURLOPT_URL,NULL);
| x=CURL_ERROR_SIZE;
| x=CURLOPT_WRITEFUNCTION;
| x=CURLOPT_WRITEDATA;
| x=CURLOPT_ERRORBUFFER;
| x=CURLOPT_STDERR;
| x=CURLOPT_VERBOSE;
| if (x) {;}
|
|   ;
|   return 0;
| }
configure:3759: result: no
configure:3852: error: libcurl development files required
```

**Comment by giordano** [comment]
You want to try to compile the same file with
```
cc -o conftest -g -O2 -I/workspace/destdir/include -I/workspace/destdir/include  conftest.c -L/workspace/destdir/lib -lcurl -Wl,--trace
```
to see what library is being linked to.  It's possible that you're picking up a system library.

**Comment by fingolfin** [comment]
It definitely is picking up the right `libcurl.so`. I did `apk del curl` and also added `-L$libdir` to `LDFLAGS`. Now seeing this:
```
configure:3644: result: 8.11.1
configure:3707: checking whether libcurl is usable
configure:3742: cc -o conftest -g -O2 -I/workspace/destdir/include   conftest.c -L/workspace/destdir/lib -lcurl  >&5
/workspace/destdir/lib/libcurl.so: undefined reference to `nghttp2_option_set_no_rfc9113_leading_and_trailing_ws_validation'
```

Also:
```
sandbox:${WORKSPACE}/srcdir/curlInterface-2.4.0 # find / -name libcurl.so
find: ‘/proc/tty/driver’: Permission denied
/workspace/x86_64-linux-musl-libgfortran4-cxx11/artifacts/50b0a37e3c68111eaf6a18eb9c31d9a63e4ad407/lib/libcurl.so
/workspace/x86_64-linux-musl-libgfortran4-cxx11/destdir/lib/libcurl.so
sandbox:${WORKSPACE}/srcdir/curlInterface-2.4.0 # find / -name curl.h
find: ‘/proc/tty/driver’: Permission denied
/workspace/x86_64-linux-musl-libgfortran4-cxx11/artifacts/50b0a37e3c68111eaf6a18eb9c31d9a63e4ad407/include/curl/curl.h
/workspace/x86_64-linux-musl-libgfortran4-cxx11/destdir/include/curl/curl.h
```

**Comment by fingolfin** [comment]
Investigated the libcurl artifacts at https://github.com/JuliaBinaryWrappers/LibCURL_jll.jl/releases for `x86-linux-musl` and `x86-linux-gnu`. Both actually reference `nghttp2_option_set_no_rfc9113_leading_and_trailing_ws_validation`, i.e.:
```
$ nm libcurl.so.4.8.0 | fgrep nghttp2_option_set_no_rfc9113_leading_and_trailing_ws_validation
                 U nghttp2_option_set_no_rfc9113_leading_and_trailing_ws_validation
```
for both of them.

For both `ldd` shows the same list of libs:
```
$ ldd libcurl.so.4.8.0
	linux-vdso.so.1 (0x00007ffe1f5f2000)
	libnghttp2.so.14 => /lib/x86_64-linux-gnu/libnghttp2.so.14 (0x00007f8c5c029000)
	libssh2.so.1 => /lib/x86_64-linux-gnu/libssh2.so.1 (0x00007f8c5bffa000)
	libssl.so.3 => not found
	libcrypto.so.3 => not found
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f8c5bfde000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f8c5bfd4000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f8c5bfaf000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f8c5bdbd000)
	libgcrypt.so.20 => /lib/x86_64-linux-gnu/libgcrypt.so.20 (0x00007f8c5bc9f000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f8c5c330000)
	libgpg-error.so.0 => /lib/x86_64-linux-gnu/libgpg-error.so.0 (0x00007f8c5bc7c000)
```


On https://github.com/curl/curl/issues/11065 someone mentions this:

> This is not a curl bug. This is libcurl having been built with an nghttp2 version >= 1.50.0 (which has that function) and then at run-time it uses an older nghttp2 library that doesn't have the function.

File: G/GAP_pkg/GAP_pkg_curlinterface/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+sources = [
+    ArchiveSource("https://github.com/gap-packages/curlInterface/releases/download/v$(upstream_version)/curlInterface-$(upstream_version).tar.gz",
+                  "6f758ad512edf033ba8892875c3216cf111feb5b856909b84889cad89c78a4ff"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd curlInterface*
+
+apk del curl #
```
Comment (fingolfin) [inline G/GAP_pkg/GAP_pkg_curlinterface/build_tarballs.jl:22]: even am trying to remove the Alpine version of `curl`

File: G/GAP_pkg/GAP_pkg_curlinterface/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd curlInterface*
+
+apk del curl  # make sure we don't link against system's libcurl by accident
+
+#export CPPFLAGS="-I$includedir"  # workaround issue with clang on Linux with musl libc, see https://github.com/JuliaPackaging/Yggdrasil/pull/10000
+#export LDFLAGS="-L$libdir"
```
Comment (fingolfin) [inline G/GAP_pkg/GAP_pkg_curlinterface/build_tarballs.jl:25]:
```suggestion
```

File: G/GAP_pkg/GAP_pkg_curlinterface/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+cp bin/*/*.so ${prefix}/lib/gap/
+
+install_license LICENSE
+"""
+
+name = gap_pkg_name(name)
+platforms, dependencies = setup_gap_package(gap_version, gap_lib_version)
+
+append!(dependencies, [
+    Dependency("LibCURL_jll"; compat="7.73,8"),
```
Comment (fingolfin) [inline G/GAP_pkg/GAP_pkg_curlinterface/build_tarballs.jl:36]: Solution was to add this `compat` bound, thanks to @giordano for pointing that out!

**Pull request merged**